scripts/screeps_tencent_batch_rl_runner.py

```
@@ -1117,6 +1130,33 @@ def safe_extract_tar(tar_path: Path, extract_dir: Path) -> None:
                 shutil.copyfileobj(source, out)
 
 
+def validated_remote_policy_update(raw: Any, top_level_safety: dict[str, Any]) -> Any:
+    unsafe = unsafe_policy_update_safety_flags(raw, "policyUpdate", top_level_safety)
+    if unsafe:
+        raise BatchRunError("remote policyUpdate safety flags are unsafe: " + "; ".join(unsafe))
+    return copy.deepcopy(raw)
+
+
+def unsafe_policy_update_safety_flags(value: Any, label: str, top_level_safety: dict[str, Any]) -> list[str]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)